src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.LocalFunctions.cs

```diff
@@ -133,7 +133,15 @@ public override BoundNode VisitLocalFunctionStatement(BoundLocalFunctionStatemen
             var oldReads = usages.ReadVars;
             usages.ReadVars = BitVector.Empty;
 
-            if (!localFunc.WasCompilerGenerated) EnterParameters(localFuncSymbol.Parameters);
+            if (!localFunc.WasCompilerGenerated)
+            {
+                EnterParameters(localFuncSymbol.Parameters);
+
+                foreach (var parameter in localFuncSymbol.Parameters)
+                {
+                    _variableUsePass.Visit(parameter.ExplicitDefaultExpression);
```

Comment:
>_variableUsePass [](start = 20, length = 16)

It looks like we are resetting the recursion depth to zero, I believe DataFlowPass keeps its own recursion depth for the same purpose as the _variableUsePass. When we visit default values here we should start with the current recursion depth of the DataFlowPass, there should be examples how to do this in the code base.